graspologic/simulations/simulations.py

```diff
@@ -795,3 +796,187 @@ def p_from_latent(X, Y=None, rescale=False, loops=True):
+
+    >>> rng = np.random.default_rng(1)
+    >>> np.random.seed(1)
+    >>> alpha = [0.05, 0.05]
+    >>> mmsbm(n, p, alpha, rng = rng)
+    array([[0., 1., 0., 0., 0., 0.],
+           [1., 0., 0., 0., 0., 0.],
+           [0., 0., 0., 0., 0., 0.],
+           [0., 0., 0., 0., 1., 1.],
+           [0., 0., 0., 1., 0., 1.],
+           [0., 0., 0., 1., 1., 0.]])
+
+    """
+
+    check_scalar(x=n, name="n", target_type=int, min_val=1)
+
+    p = check_array(p, ensure_2d=True)
+    nx, ny = p.shape
+    if nx != ny:
+        msg = "p must be a square matrix, not {}".format(p.shape)
+        raise ValueError(msg)
+    if not np.issubdtype(p.dtype, np.number):
+        msg = "There are non-numeric elements in p"
+        raise ValueError(msg)
+    if np.any(p < 0) or np.any(p > 1):
+        msg = "Values in p must be in between 0 and 1."
+        raise ValueError(msg)
+
+    alpha = check_array(alpha, ensure_2d=False, ensure_min_features=1)
+    if not np.issubdtype(alpha.dtype, np.number):
+        msg = "There are non-numeric elements in alpha"
+        raise ValueError(msg)
+    if np.any(alpha <= 0):
+        msg = "Alpha entries must be > 0."
+        raise ValueError(msg)
+    if alpha.shape != (len(p),):
+        msg = "alpha must be a list or np.array of shape {c}, not {w}.".format(
+            c=(len(p),), w=alpha.shape
+        )
+        raise ValueError(msg)
+
+    if not isinstance(rng, (np.random._generator.Generator)):
+        msg = "rng must be <class 'numpy.random._generator.Generator'> not {}.".format(
+            type(rng)
+        )
+        raise TypeError(msg)
+    elif rng == None:
+        rng = np.random.default_rng()
+
+    if type(loops) is not bool:
+        raise TypeError("loops is not of type bool.")
+    if type(directed) is not bool:
+        raise TypeError("directed is not of type bool.")
+    if type(return_labels) is not bool:
+        raise TypeError("return_labels is not of type bool.")
+
+    if not directed:
+        if np.any(p != p.T):
+            raise ValueError("Specified undirected, but P is directed.")
+
+    # Naming convention follows paper listed in references.
+    mm_vectors = rng.dirichlet(alpha, n)
+
+    mm_vectors = np.array(sorted(mm_vectors, key=lambda x: np.argmax(x)))
+
+    # labels:(n,n) matrix with all membership indicators for initiators and receivers
+    # instead of storing the indicator vector, argmax is directly computed
+    # check docstrings for more info.
+    labels = np.apply_along_axis(
+        lambda p_vector: np.argmax(
+            rng.multinomial(n=1, pvals=p_vector, size=n), axis=1
+        ),
+        axis=1,
+        arr=mm_vectors,
+    )
+
+    P = np.zeros((n, n))
+    for i in range(n):
+        for j in range(n):
+            P[i, j] = p[labels[i, j], labels[j, i]]
+
+    A = sample_edges(P, directed=directed, loops=loops)
+
+    if not loops:
+        np.fill_diagonal(labels, -1)
```

Comment:
```suggestion
        np.fill_diagonal(labels, 0)
```

Reply:
Thank you Ben for the prompt review of the changes made! 

 I wanted to have some type of noticeable difference like a N/A assigned to the diagonal so that it was even more clear that a zero on the diagonal does not mean, in a graph with no loops, that node i when interacting with itself is assigned to community one(which is denoted by index 0 in the labels array), but rather that no such assignment is done in the first place because no edge is sampled. 

Is there a better way of doing this than to assign -1? 

If you would still prefer that I change them all to zero then maybe I can explain in the docstring that a zero on the diagonal in labels if the graphs has loops indicates that no edge was sampled. 

Alternatively, when I return labels I could add 1 to all the indexes so that community 1 is denoted with 1, community 2 with 2 etc. 

Please let me know what you would prefer between these options. I am happy with making whichever of these changes

Reply:
ah i see i misunderstood. yes perhaps np.nan is more clear